subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmVariantBuilder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.jvm;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.internal.HasInternalProtocol;
+
+/**
+ * A Java component builder, allowing the automatic creation of a number of configurations,
+ * tasks, ...
+ *
+ * @since 6.7
+ */
+@Incubating
+@HasInternalProtocol
+public interface JvmVariantBuilder {
```

Comment:
What bout `withTests()`? (to create a test task for the source set)

Reply:
I didn't want to go to the task level yet, especially for tests since we'd like to introduce matrix testing.

Reply:
Lets do this later